src/index.ts

```diff
@@ -103,16 +103,14 @@ export class Server extends EventEmitter {
   protected connected: boolean
   protected connectionOpts: ConnectionOpts
 
-  constructor (opts: ServerOpts) {
+  constructor ({ serverSecret, ...opts }: ServerOpts) {
     super()
-    this.serverSecret = opts.serverSecret || randomBytes(32)
+    this.serverSecret = serverSecret || randomBytes(32)
```

Comment:
Can't the serverSecret stay as part of the opts here? so these 2 lines do not need to be changed?

Reply:
I decided to leave it in because it simplifies the connectionOpts below so `serverSecret` doesn't need to be overwritten as undefined, but if you'd prefer I can change it back